compiler/rustc_middle/src/ty/query.rs

```diff
@@ -102,14 +102,28 @@ impl<'tcx> TyCtxt<'tcx> {
     }
 }
 
-/// Helper for `TyCtxtEnsure` to avoid a closure.
-#[inline(always)]
-fn noop<T>(_: &T) {}
-
-/// Helper to ensure that queries only return `Copy` types.
-#[inline(always)]
-fn copy<T: Copy>(x: &T) -> T {
-    *x
+fn evaluate_query<'tcx, Cache>(
```

Comment:
Did you try this with `#[inline(always)]`? The callers all have that so this should regress without it.